### PR TITLE
BUG: byte math errors for decoding bitmap pngs

### LINKED ISSUE #535: Bitmap PNGs are not decoded
The current implementation of FlateDecode.decode assumes the PNG is encoded with `BitsPerComponent == 8`.  This is not valid for bitmap PNGs.  In this case, each row is still prefixed with the filter byte then followed by the bits, padded out to the final byte.

Pull request coming shortly.

### PR DESCRIPTION
**Pull request by chrisegner**
Fixes #535

### CODE REVIEW

**Comment by chrisegner** [comment]
Oops - munged the commit a bit while cleaning up git. Fixed in 07f2dea.

File: PyPDF2/filters.py [hunk 1]
```diff
@@ -44,12 +44,14 @@
...
 if version_info < ( 3, 0 ):
-    from cStringIO import StringIO
+    from cStringIO import StringIO as BytesIO
+    bytearray = buffer
 else:
     from io import StringIO
 
 import struct
 
+
```
Comment (MartinThoma) [inline PyPDF2/filters.py:54]:
```suggestion
```

File: PyPDF2/filters.py [hunk 2]
```diff
@@ -44,7 +44,8 @@
 from .utils import PdfReadError, ord_, paethPredictor
 
 if version_info < ( 3, 0 ):
-    from cStringIO import StringIO
+    from cStringIO import StringIO as BytesIO
+    bytearray = buffer
```
Comment (MartinThoma) [inline PyPDF2/filters.py:48]:
```suggestion
    bytearray = buffer  # noqa: F821
```

**Comment by MartinThoma** [comment]
Sadly, this part is still untested. Would you mind adding a unit test? Or do you have an example PDF that triggers this code, then I can take care of it

**Pull request closed without merging**